docs/configure.md

```diff
@@ -0,0 +1,153 @@
+              "privileges": [
+                "manage",
+                "index",
+                "create_index",
+                "create"
+              ]
+            }
+          ]
+        }
+      }
+    }
+    ```
+
+The API Key will be returned in the response. Save the `encoded` field for use in the configuration.
+
+Read the [Elasticsearch documentation](https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-create-api-key.html) for more information on creating API Keys.
+
+## Add the integration
+
+This component is configured interactively via Home Assistant's integration configuration page.
+
+1. Verify you have restarted Home Assistant after installing the component.
+2. From the [`Integrations` configuration menu](https://my.home-assistant.io/redirect/integrations/), add a new `Elasticsearch` integration.
+3. Select the appropriate authentication method
+4. Provide connection information and credentials (if using) to begin setup.
+5. Once the integration is setup, you may tweak all settings via the `Configure` button on the [integrations page](https://my.home-assistant.io/redirect/integration/?domain=elasticsearch){:target="_blank"}.
+
+## Configuration options
+
+Select `Configure` from the integration's homepage to configure the following settings.
+
+[![Open your Home Assistant instance and show the Elasticsearch integration.](https://my.home-assistant.io/badges/integration.svg)](https://my.home-assistant.io/redirect/integration/?domain=elasticsearch){:target="_blank"}
+
+### Publish events to Elasticsearch
+Indicates if events should be published to Elasticsearch. If disabled, no events will be published to Elasticsearch. The default is `enabled`.
+
+### Publish frequency
+The frequency at which events are published to Elasticsearch, in seconds. The default is `60`.
+
+### Publish mode
+There are three modes to publish data to Elasticsearch:
+
+- `All` - Publish configured entities to Elasticsearch, including those which did not undergo a state or attribute change.
+- `State changes` - Publish configured entities to Elasticsearch only when their state changes.
+- `Any changes` - Publish configured entities to Elasticsearch when their state or attributes change.
+
+| Publish Mode | State Change | Attribute Change | No Change |
+| ---- | ---- | ---- | ---- |
+| All | ✅  Publishes | ✅ Publishes | ✅ Publishes |
+| Any Changes | ✅  Publishes | ✅  Publishes | 🚫 Does not publish |
+| State Changes | ✅  Publishes | 🚫 Does not publish | 🚫 Does not publish |
+
+### Entity selection
+You can choose to include/exclude specific entities and/or entire domains. This allows you to publish only the entities you are interested in to Elasticsearch.
+By default, all entities and domains are included.
+You can combine inclusion and exclusion filters to fine-tune the entities you want to publish.
+
+The following flowchart describes the logic used to determine if an entity is published:
+
+```mermaid
+graph LR
```

Comment:
see https://github.com/legrego/homeassistant-elasticsearch/issues/273 -- I think we should change this logic

Reply:
Let's continue the discussion in #273. I'll adjust this diagram once we settle on an implementation.